scripts/zkai_stwo_transformer_block_plan.py

```
@@ -0,0 +1,231 @@
+    if target.get("statement_kind") != TARGET_STATEMENT_KIND:
+        raise PlanValidationError(f"target.statement_kind must be {TARGET_STATEMENT_KIND!r}")
+    if target.get("width") != 4:
+        raise PlanValidationError("target.width must remain the bounded width-4 implementation target")
+
+    operations = _required_list(target.get("operations"), "target.operations")
+    operation_ids = _ids(operations, "target.operations")
+    _require_superset(operation_ids, REQUIRED_OPERATION_IDS, "target.operations")
+
+    public_commitments = set(_required_list(target.get("public_commitments"), "target.public_commitments"))
+    if not all(isinstance(item, str) and item for item in public_commitments):
+        raise PlanValidationError("target.public_commitments must contain only non-empty strings")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)